iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/pipeconsensus/PipeConsensusAsyncConnector.java

```diff
@@ -478,6 +500,13 @@ public synchronized void clearRetryEventsReferenceCount() {
     }
   }
 
+  public synchronized void clearTransferBufferReferenceCount() {
```

Comment:
There may still be some events added into the "transferBuffer" even if this function has finished.. Can use "synchronized" or the similar handling logic to "addFailureEventToRetryQueue".

Reply:
good catch, fixed